### DISCUSSION TITLE
Cosmos DB: Nested JSON object - GraphQL schema

### DISCUSSION THREAD

**Discussion by vrm535** [Azure/data-api-builder#2342] [General]
Hi Team:

I am trying to create DAB layer for my backend CosmosDB. Item collection is a nested JSON object. I created the dab-configuration file with just one entity that is employee and created the graphql schema as shown below.

When I am validating/starting the application, I am seeing the error as Azure.DataApiBuilder.Service.Exceptions.DataApiBuilderException: The entity 'Contact' was not found in the 
runtime config. 

Could you please guide me here how to deal with this scenario.

GraphQL- schema
=================

type Address {
    address1: String
    address2: String
    city: String
    state: Sting
}

type Phone {
   home: String
   office: String
}

type Contact {
   address: Address
   phone: Phone
}

type Employee @model {
       firstName: String
       lastName: String
       middleName: String
       contact: Contact
}

**Comment by chullybun** [comment]
I had the similar issue today too, and tried the following which seemed to work. Basically it seems, you will need to add each entity, but point them at the same source:

`dab add Employee --source "employees" --permissions "anonymous:*"`
`dab add Contact --source "employees" --permissions "anonymous:*"`

It was this [file](https://github.com/Azure/data-api-builder/blob/e8d9fcb9ac1c2aa8b47808b0a658a09e092545e0/config-generators/cosmosdb_nosql-commands.txt#L2) that led me to this.

Good luck 🙂

**Comment by sajeetharan** [comment]
@vrm535 did the above suggestion help to solve the issue?

**Comment by sezal98** [comment]
Hi @vrm535 
Please update in case you face it again. Closing the issue for now
